[Doc] better specification of the behaviour of in operator in various contexts (#11967)
elastic/logstash: docs/static/configuration.asciidoc

}
----------------------------------

> You can use the `in` operator to test whether a field contains a specific string, key, or (for lists) element.

Reviewer: ```suggestion
You can use the `in` operator to test whether a field contains a specific string, key, or list element.
```

> It has to be noted that the semantic meaning of `in` could vary, based on the target type. For example if applied to
> string means "is substring of" while if applied to a collection type means "collection exactly contains the value".

Reviewer: ```suggestion
Note that the semantic meaning of `in` can vary, based on the target type. For example, when applied to
a string. `in` means "is a substring of." When applied to a collection type, `in` means "collection contains the exact value."
```
